File: codegen-tool/src/main/java/com/dqjq/codegen/Generator/ServiceImplGenerator.java
package com.dqjq.codegen.Generator;

import com.dqjq.codegen.CodegenConfig;
import com.dqjq.codegen.CodegenFieldDef;
import lombok.Getter;

import java.io.IOException;
import java.nio.file.Path;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * Service 实现类生成器
 * <p>
 * 负责生成业务逻辑实现类。
 * </p>
 */
@Getter
public class ServiceImplGenerator extends BaseGenerator {

    private Path implPath;

    public ServiceImplGenerator(CodegenConfig cfg) {
        super(cfg);
    }

    @Override
    public void generate() throws IOException {
        Path baseOut = getBaseOutputDir();
        String basePackagePath = cfg.packageBase.replace('.', '/');
        String modulePath = baseOut + "/" + basePackagePath + "/" + cfg.module;
        Path implDir = projectRoot.resolve(modulePath + "/service/impl");
        
        ensureDirectory(implDir);
        
        this.implPath = implDir.resolve(cfg.entityName + "ServiceImpl.java");

        String servicePkg = cfg.packageBase + "." + cfg.module + ".service";
        String implPkg = servicePkg + ".impl";
        String dtoPkg = cfg.packageBase + "." + cfg.module + ".dto";
        String entityPkg = cfg.packageBase + "." + cfg.module + ".entity";
        String repoPkg = cfg.packageBase + "." + cfg.module + ".repository";
        String mapperPkg = cfg.packageBase + "." + cfg.module + ".mapper";

        Map<String, Object> data = new HashMap<>();
        data.put("package", implPkg);
        data.put("dtoPackage", dtoPkg);
        data.put("entityPackage", entityPkg);
        data.put("repoPackage", repoPkg);
        data.put("mapperPackage", mapperPkg);
        data.put("servicePackage", servicePkg);
        data.put("entityName", cfg.entityName);
        data.put("dtoName", cfg.entityName + "Dto");
        data.put("idType", cfg.idType);
        data.put("fields", cfg.fields);

        renderFreemarker("serviceImpl.ftl", data, implPath);
    }
}

File: codegen/codegen-tool-py/codegen/generators/service_impl_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
ServiceImpl生成器
生成Java ServiceImpl实现类
"""

from pathlib import Path
from codegen.generators.base_generator import BaseGenerator

class ServiceImplGenerator(BaseGenerator):
    """ServiceImpl生成器"""
    
    def generate(self):
        """
        生成ServiceImpl实现类
        """
        # 准备上下文数据
        context = {
            'package': self.get_full_package('service.impl'),
            'entity_name': self.code_generator.entity_name,
            'id_type': self.code_generator.id_type,
            'entity_package': self.get_full_package('entity'),
            'dto_package': self.get_full_package('dto'),
            'service_package': self.get_full_package('service'),
            'repository_package': self.get_full_package('repository'),
            'mapper_package': self.get_full_package('mapper'),
            'service_impl_base_class': self.config_dict.get('serviceImplBaseClass', ''),
            'has_base_class': bool(self.config_dict.get('serviceImplBaseClass', ''))
        }
        
        # 渲染模板
        content = self.render_template('java/serviceImpl.j2', context)
        
        # 生成文件路径
        output_dir = self.code_generator.get_output_path('service/impl')
        self.output_path = output_dir / f"{self.code_generator.entity_name}ServiceImpl.java"
        
        # 写入文件
        self.write_file(self.output_path, content)
